gettsim/transfers/arbeitsl_geld.py

```diff
@@ -29,81 +30,143 @@ def arbeitsl_geld_m(
+        See :func:`anspruchsdauer_gesamt`.
+    alter
+        See basic input variable :ref:`alter <alter>`.
+    arbeitsl_geld_params
+        See params documentation :ref:`arbeitsl_geld_params <arbeitsl_geld_params>`.
+    arbeitsl_geld_berechtigt
+        See :func:`arbeitsl_geld_berechtigt`.
+    arbeitsl_geld_m_vorj
+        See basic input variable :ref:`arbeitsl_geld_m_vorj <arbeitsl_geld_m_vorj>`.
+    arbeitsl_geld_m_v2j
+        See basic input variable :ref:`arbeitsl_geld_m_v2j <arbeitsl_geld_m_v2j>`.
+
+    Returns
+    -------
+
+    """
+
     if arbeitsl_geld_berechtigt:
-        if anz_kinder_tu == 0:
-            arbeitsl_geld_satz = arbeitsl_geld_params["satz_ohne_kinder"]
+        if alter >= arbeitsl_geld_params["anwartschaftszeit"]["threshold_5"]["age"]:
+            out = max(
+                anspruchsdauer_gesamt - arbeitsl_geld_m_vorj - arbeitsl_geld_m_v2j, 0
+            )
         else:
-            arbeitsl_geld_satz = arbeitsl_geld_params["satz_mit_kindern"]
-
-        out = arbeitsl_geld_eink_vorj_proxy * arbeitsl_geld_satz
+            out = max(anspruchsdauer_gesamt - arbeitsl_geld_m_vorj, 0)
     else:
-        out = 0.0
+        out = 0
 
     return out
 
 
-def arbeitsl_monate_gesamt(
-    arbeitsl_monate_lfdj: int,
-    arbeitsl_monate_vorj: int,
-    arbeitsl_monate_v2j: int,
+def anspruchsdauer_gesamt(
+    anwartschaftszeit: int,
+    alter: int,
+    arbeitsl_geld_params: dict,
 ) -> int:
-    """Aggregate months of unemployment over the last two years.
+    """Calculate the amount of months a person could receive unemployment benifit.
 
     Parameters
     ----------
-    arbeitsl_monate_lfdj
-        See basic input variable :ref:`arbeitsl_monate_lfdj <arbeitsl_monate_lfdj>`.
-    arbeitsl_monate_vorj
-        See basic input variable :ref:`arbeitsl_monate_vorj <arbeitsl_monate_vorj>`.
-    arbeitsl_monate_v2j
-        See basic input variable :ref:`arbeitsl_monate_v2j <arbeitsl_monate_v2j>`.
+    anwartschaftszeit
+        See basic input variable :ref:`anwartschaftszeit <anwartschaftszeit>`.
+    alter
+        See basic input variable :ref:`alter <alter>`.
+    arbeitsl_geld_params
+        See params documentation :ref:`arbeitsl_geld_params <arbeitsl_geld_params>`.
 
     Returns
     -------
 
     """
-    return arbeitsl_monate_lfdj + arbeitsl_monate_vorj + arbeitsl_monate_v2j
+    for threshold in [
+        "threshold_1",
+        "threshold_2",
+        "threshold_3",
+        "threshold_4",
+        "threshold_5",
+        "threshold_6",
+        "threshold_7",
+    ]:
+        if (
+            anwartschaftszeit
+            >= arbeitsl_geld_params["anwartschaftszeit"][threshold]["min"]
+            and alter >= arbeitsl_geld_params["anwartschaftszeit"][threshold]["age"]
+        ):
+            out = arbeitsl_geld_params["anwartschaftszeit"][threshold]["anspruch"]
+        else:
+            out = 0
+
+        return out
+    return out
 
 
 def arbeitsl_geld_berechtigt(
```

Comment:
The changes in this function change the interpretation of `arbeitsl_geld_berechtigt` as subjects who have exhausted their duration of unemployment benefit will be "berechtigt" for unemployment benefit (just for 0 months only). Is there a specific reason for that? Otherwise, I would stick to the old interpretation that those subjects would be counted as not eligible.